studio/src/components/Timeline/DetailsList/TimelineDetailsList.tsx

```diff
@@ -0,0 +1,71 @@
+import { isMizuOrphanLog } from "@/queries";
+import {
+  type Waterfall,
+  cn,
+  isFetchSpan,
+  isIncomingRequestSpan,
+} from "@/utils";
+import { memo } from "react";
+import { useTimelineContext } from "../context";
+import { OrphanLog } from "./OrphanLog";
+import { FetchSpan, GenericSpan, IncomingRequest } from "./spans";
+
+function TimelineListDetailsComponent({
+  waterfall,
+}: {
+  waterfall: Waterfall;
+}) {
+  const { highlightedSpanId, setHighlightedSpanId } = useTimelineContext();
+  // TODO: merge spans and orphanLogs
+  return (
+    <div className="grid gap-2" id="trace-details-v2">
+      {waterfall.map((item) => (
+        <div
+          key={getId(item)}
+          onMouseEnter={() => setHighlightedSpanId(getId(item))}
+          onMouseLeave={() => setHighlightedSpanId(null)}
+          className={cn(
+            "p-2 max-w-full overflow-hidden",
+            "border-l-2 border-transparent transition-all bg-transparent",
+            "hover:bg-primary/10 hover:border-blue-500",
+            "data-[highlighted=true]:bg-primary/10",
+            "relative after:absolute after:bottom-[-4px] after:bg-muted-foreground/30 after:w-full after:h-px last:after:h-0",
```

Comment:
nit: not wild about the border around the timeline. might be good to get some ideas from simon on how to tweak this